### ISSUE TITLE
Error when importing @prb/math packages when installed via npm

### ISSUE THREAD

**Issue by waynehoover** [sablier-labs/v2-core#646]
After adding `@sablier/core-v2` to my package.json I'm getting this error when I compile my contract:

`File @prb/math/SD59x18.sol, imported from @sablier/v2-core/src/types/Math.sol, not found.`

My contract imports look like this:
```
import { ISablierV2LockupLinear } from "@sablier/v2-core/src/interfaces/ISablierV2LockupLinear.sol";
import { ud60x18 } from "@sablier/v2-core/src/types/Math.sol";
import { Broker, LockupLinear } from "@sablier/v2-core/src/types/DataTypes.sol";
import { IERC20 } from "@sablier/v2-core/src/types/Tokens.sol";
```

package.json:
```
...
"@sablier/v2-core": "^1.0.1",
...
```

This is because the contracts when deployed to the @prb/math npm package are in the `src` folder as you can see here: https://www.npmjs.com/package/@prb/math?activeTab=code

(the same is true for the @sablier/v2-core repo, but that's okay because I can manually add the src/ directory to imports there)

So I suppose a solution would be to move the contracts out of the src folder when deploying the package to npm, this would be backwards compatible with forge, but not with anyone including @prb/math with npm and referencing it from the src directory.

If there is a way to manually map the src folder in npm (like you can with forge's remappings.txt) that would also work, but I didn't find any solutions like that with a quick google.

-Update-
I was able to get this working using the [hardhat-foundry](https://hardhat.org/hardhat-runner/docs/advanced/hardhat-and-foundry) plugin and installing @prb/math the foundry way. But ideally, users should be able to use this repo without foundry.

**Comment by PaulRBerg** [comment]
Thanks for the detailed report @waynehoover. I'm sorry to hear you bumped into all these issues.

This may be related to this Hardhat bug:

https://github.com/NomicFoundation/hardhat/issues/4194

I agree that Hardhat users should not have to install third-party dependencies via Foundry. It's bad DX. Hopefully once Hardhat fixes the bug above, you won't have to do this anymore.

Side note: we are considering to move away from remappings to Node.js deps.

**Comment by PaulRBerg** [comment]
hey @waynehoover! It turns out that there was another bug in Foundry that was getting in the way of making the remappings work:

https://github.com/foundry-rs/foundry/issues/5540

They fixed it in the meantime - if you run `foundryup`, do you notice anything different now?

**Comment by PaulRBerg** [comment]
Thank you, @waynehoover. The extra details are helpful.

Using [`hardhat-foundry`](https://hardhat.org/hardhat-runner/docs/advanced/hardhat-and-foundry) seems the way to go for the time being.

We will likely switch to using npm packages everywhere in the long term since it seems like the most robust solution. I wouldn't want to mess around with the location of the contract files at build time since that would add some maintenance overhead. Similarly, copying over dependencies would require us to keep our copy of PRBMath in sync with the actual, latest PRBMath, and it would also be a bit self-defeating.

**Comment by PaulRBerg** [comment]
As shown by this [conversation](https://github.com/allo-protocol/allo-v2/pull/158#issuecomment-1675172219), this issue with the `@prb/math` remappings is becoming increasingly more important.

I have marked this as "priority0". @andreivladbrg, it would be helpful if you could pick up this task during your next sprint; basically, the solution is to swap all import paths to contain `src/`, and update the paths in `remappings.txt` accordingly.

**Comment by PaulRBerg** [comment]
Hey @waynehoover, glad to let you know that we just shipped v1.0.2 of the [`@sablier/v2-core`](https://www.npmjs.com/package/@sablier/v2-core) package on npm, which should address all of the import paths bug you bumped into.

Let us know if it works!
